core/server/src/main/java/tachyon/web/WebInterfaceMasterMetricsServlet.java

```diff
@@ -0,0 +1,138 @@
+
+package tachyon.web;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.TreeMap;
+
+import javax.annotation.concurrent.ThreadSafe;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.MetricRegistry;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.master.MasterContext;
+import tachyon.master.MasterSource;
+import tachyon.master.TachyonMaster;
+import tachyon.util.CommonUtils;
+
+/**
+ * Servlet that provides data for viewing the master metrics values
+ */
+@ThreadSafe
+public final class WebInterfaceMasterMetricsServlet extends WebInterfaceAbstractMetricsServlet {
+
+  private static final long serialVersionUID = -1481253168100363787L;
+  private final transient TachyonMaster mMaster;
+
+  /**
+   * Creates a new instance of {@link WebInterfaceMasterMetricsServlet}.
+   *
+   * @param master Tachyon master
+   */
+  public WebInterfaceMasterMetricsServlet(TachyonMaster master) {
+    super();
+    mMaster = Preconditions.checkNotNull(master);
+  }
+
+  /**
+   * Redirects the request to a JSP after populating attributes via populateValues.
+   *
+   * @param request the {@link HttpServletRequest} object
+   * @param response the {@link HttpServletResponse} object
+   * @throws ServletException if the target resource throws this exception
+   * @throws IOException if the target resource throws this exception
+   */
+  @Override
+  protected void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+    populateValues(request);
+    getServletContext().getRequestDispatcher("/metrics.jsp").forward(request, response);
+  }
+
+  @Override
+  protected void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+    populateValues(request);
+    getServletContext().getRequestDispatcher("/metrics.jsp").forward(request, response);
+  }
+
+  /**
+   * Populates key, value pairs for UI display.
+   *
+   * @param request The {@link HttpServletRequest} object
+   * @throws IOException if an I/O error occurs
+   */
+  private void populateValues(HttpServletRequest request) throws IOException {
+    MetricRegistry mr = mMaster.getMasterMetricsSystem().getMetricRegistry();
+
+    MasterSource masterSource = MasterContext.getMasterSource();
+
+    Long masterCapacityTotal = (Long) mr.getGauges()
+        .get(CommonUtils.argsToString(".", masterSource.getName(), MasterSource.CAPACITY_TOTAL))
+        .getValue();
+    Long masterCapacityUsed = (Long) mr.getGauges()
+        .get(CommonUtils.argsToString(".", masterSource.getName(), MasterSource.CAPACITY_USED))
+        .getValue();
+
+    int masterCapacityUsedPercentage = (int) (100L * masterCapacityUsed / masterCapacityTotal);
```

Comment:
I'm afraid when masterCapacityUsed is 0, an exception is thrown at this line. As you said, it's possible users may misconfigure things, but it'll be bad to crash, instead of displaying 0 or sth else.